src/NHibernate/Type/CollectionType.cs

```diff
@@ -161,7 +161,7 @@ public override object Disassemble(object value, ISessionImplementor session, ob
 		public override void BeforeAssemble(object oid, ISessionImplementor session)
 		{
 			var queryCacheQueue = session.PersistenceContext.BatchFetchQueue.QueryCacheQueue;
-			if (queryCacheQueue == null)
+			if (queryCacheQueue == null || oid == null)
```

Comment:
Can you please add oid null check before retrieving `queryCacheQueue`

Reply:
Done.